src/main/java/tech/fastj/systems/audio/AudioEventListener.java

```diff
@@ -23,33 +26,51 @@
 
     private final Audio audio;
 
+
     private static final Map<LineEvent.Type, BiConsumer<LineEvent, AudioEventListener>> AudioEventProcessor = Map.of(
             LineEvent.Type.OPEN, (audioEvent, audioEventListener) -> audioEventListener.audioOpenAction.accept(audioEvent),
             LineEvent.Type.START, (audioEvent, audioEventListener) -> {
-                switch (audioEventListener.audio.getPreviousPlaybackState()) {
+                PlaybackState previousPlaybackState = audioEventListener.audio.getPreviousPlaybackState();
+                switch (previousPlaybackState) {
                     case Paused: {
                         audioEventListener.audioResumeAction.accept(audioEvent);
                     }
                     case Stopped: {
                         audioEventListener.audioStartAction.accept(audioEvent);
                         break;
                     }
+                    default: {
+                        exceptionOnPlaybackState(previousPlaybackState);
+                    }
                 }
             },
             LineEvent.Type.STOP, (audioEvent, audioEventListener) -> {
-                switch (audioEventListener.audio.getCurrentPlaybackState()) {
+                PlaybackState currentPlaybackState = audioEventListener.audio.getCurrentPlaybackState();
+                switch (currentPlaybackState) {
                     case Paused: {
                         audioEventListener.audioPauseAction.accept(audioEvent);
                     }
                     case Stopped: {
                         audioEventListener.audioStopAction.accept(audioEvent);
                         break;
                     }
+                    default: {
+                        exceptionOnPlaybackState(currentPlaybackState);
+                    }
                 }
             },
             LineEvent.Type.CLOSE, (audioEvent, audioEventListener) -> audioEventListener.audioCloseAction.accept(audioEvent)
     );
 
+    /**
+     * Throws IllegalStateException when PlaybackState passes through default cases in switch statements in
+     * AudioEventProcessor
+     * @param p the playback state
+     */
+    private static void exceptionOnPlaybackState(PlaybackState p) {
```

Comment:
I appreciate the effort (and the abstraction of the state variable -- nice), but I don't think there's much need for a specific method just to throw that specific exception. It's a nice idea, but then that leads to having a whole bunch of abstractions all over the library with the same concept. 
That, and it clutters the stack trace if the error is actually thrown! There doesn't really need to be a layer of abstraction just for exception throwing, on top of what's already there.

Reply:
Hmm that's true. I'll get rid of it!